doc/intl.md

```diff
@@ -0,0 +1,192 @@
+# Overview
+
+This document describes the current state of Android implementation of Intl APIs(formally ECMA-402 specification) in Hermes JavaScript engine. ECMA-402 is still evolving and the latest iteration is [7th edition](https://402.ecma-international.org/7.0/) which was published in June 2020. Each new edition is built on top of the last one and typically adds new capabilities typically as,
+- New Intl service constructors (for e.g. Intl.Collator, Intl.NumberFormat etc...) or extending existing ones by accepting more parameters
+- New functions or properties in the Intl object prototypes (for e.g. Intl.Collator.prototype.compare)
+- New locale aware functions in standard Javascript object prototypes (for e.g. String.prototype.localeCompare)
+
+One standard internationalization implementation strategy followed by other engines, is to bundle a library and data (typically [icu](http://site.icu-project.org/)) along with the application package. This guarantees deterministic functionality at the cost of applications package bloat.  We decided to consume the Android platform libraries and data for space efficiency, but at the cost of some variance in behaviour across  Android platforms.
+
+# ECMA-402 Compliance

```

Comment:
❤️

Reply:
Ah .. I thought about your question last week .. but missed actually replying to it ! 

I am hoping we will continue to keep the docx as a richer detailed documentation for the Intl developers and the markdown as the quick reference for Intl consumers .. I see that many aspects of our representations in the docx can be useful for the consumers too, but will also also make the contents complicated .. I think we could work on it as part of our next wave of Intl feature developement ..